src/tests/integration/test_pull_contracts.ts

```diff
@@ -0,0 +1,151 @@
+import {
+    SimbaConfig,
+    allContracts,
+} from "@simbachain/web3-suites";
+import pull from "../../tasks/contract/pullcontract";
+import {
+    FileHandler,
+} from "../tests_setup/file_handler"
+import * as fs from "fs";
+import { expect } from 'chai';
+import 'mocha';
+import * as path from 'path';
+import {cwd} from 'process';
+
+describe('testing pulling .sol file from designID', () => {
+    it('should exist in /contracts/simbaimports/ after', async () => {
+        let simbaDir = path.join(cwd(), "contracts", "SimbaImports");
+        const contractName = "TestContractVT3";
+        const oldContractID = "cb3ad592-1ca2-43b3-a9d0-cd0d0f127b32";
+        const filePath = path.join(simbaDir, `${contractName}.sol`);
+        const simbaConfig = new SimbaConfig();
+        const authStore = await simbaConfig.authStore();
+        FileHandler.removeDirectory(simbaDir);
+        await authStore!.performLogin(false);
+        let exists = fs.existsSync(filePath);
+        expect(exists).to.equal(false);
+        await pull(oldContractID);
+        exists = fs.existsSync(filePath);
+        expect(exists).to.equal(true);
+        FileHandler.removeDirectory(simbaDir);
+    }).timeout(15000);
+});
+
+describe('testing pulling source code to simba.json', () => {
+    it('source code should be in simba.json after function calls', async () => {
+        const contractName = "TestContractVT3";
+        const simbaConfig = new SimbaConfig();
+        const authStore = await simbaConfig.authStore();
+        const originalSimbaJson = SimbaConfig.ProjectConfigStore.all;
+        SimbaConfig.resetSimbaJson(originalSimbaJson, null, true);
+        let contractsInfo = SimbaConfig.ProjectConfigStore.get("contracts_info");
+        let currentKeysLength = Object.keys(contractsInfo).length
+        expect(currentKeysLength).to.eq(0);
+
+
+        await authStore!.performLogin(false);
+        // now set one contract's source code in simba.json:
+        await pull(undefined, contractName, true);
+        contractsInfo = SimbaConfig.ProjectConfigStore.get("contracts_info");
+        let entry = contractsInfo[contractName];
+        expect(entry).to.not.eq(undefined);
+        expect(entry).to.not.eq(null);
+
+        // reset
+        SimbaConfig.ProjectConfigStore.clear();
+        SimbaConfig.ProjectConfigStore.set(originalSimbaJson);
+    }).timeout(120000);
```

Comment:
These timeout a too long. If it's not done in 10-15 seconds, there should be a problem no?

Reply:
resolved in conversation.